inference-engine/src/mkldnn_plugin/nodes/mkldnn_priorbox_clustered_node.cpp

```diff
@@ -0,0 +1,178 @@
+    const int H = in_data[0];
+    const int W = in_data[1];
+    const auto output = static_cast<size_t>(4 * H * W * number_of_priors);
+    return {{2, output}};
+}
+
+bool MKLDNNPriorBoxClusteredNode::needPrepareParams() const {
+    return false;
+}
+
+void MKLDNNPriorBoxClusteredNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    addSupportedPrimDesc(
+            {{LayoutType::ncsp, Precision::I32}, {LayoutType::ncsp, Precision::I32}},
+            {{LayoutType::ncsp, Precision::FP32}},
+            impl_desc_type::ref_any);
+}
+
+void MKLDNNPriorBoxClusteredNode::createPrimitive() {
+    if (inputShapesDefined()) {
+        if (needPrepareParams())
+            prepareParams();
+        updateLastInputDims();
+    }
+}
+
+void MKLDNNPriorBoxClusteredNode::execute(mkldnn::stream strm) {
+    const int* in_data = reinterpret_cast<int*>(getParentEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const int layer_height = in_data[0];
+    const int layer_width = in_data[1];
+
+    const int* in_image = reinterpret_cast<int*>(getParentEdgeAt(1)->getMemoryPtr()->GetPtr());
+    int img_height = in_image[0];
+    int img_width = in_image[1];
+
+    float step_w = step_widths == 0 ? step : step_widths;
+    float step_h = step_heights == 0 ? step : step_heights;
+    if (step_w == 0 && step_h == 0) {
+        step_w = static_cast<float>(img_width) / layer_width;
+        step_h = static_cast<float>(img_height) / layer_height;
+    }
+
+    float* dst_data = reinterpret_cast<float*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+    const auto& out_shape = getChildEdgeAt(0)->getMemory().GetShape().getStaticDims();
+
+    size_t var_size = variances.size();
+    parallel_for(layer_height, [&](int64_t h) {
+        for (int64_t w = 0; w < layer_width; ++w) {
+            float center_x = (w + offset) * step_w;
+            float center_y = (h + offset) * step_h;
+
+            for (size_t s = 0; s < number_of_priors; ++s) {
+                float box_width = widths[s];
+                float box_height = heights[s];
+
+                float xmin = (center_x - box_width / 2.0f) / img_width;
+                float ymin = (center_y - box_height / 2.0f) / img_height;
+                float xmax = (center_x + box_width / 2.0f) / img_width;
+                float ymax = (center_y + box_height / 2.0f) / img_height;
+
+                if (clip) {
+                    xmin = (std::min)((std::max)(xmin, 0.0f), 1.0f);
+                    ymin = (std::min)((std::max)(ymin, 0.0f), 1.0f);
+                    xmax = (std::min)((std::max)(xmax, 0.0f), 1.0f);
+                    ymax = (std::min)((std::max)(ymax, 0.0f), 1.0f);
+                }
+
+                auto get_idx = [&](uint64_t cnt) -> uint64_t {
+                    return h * layer_width * number_of_priors * cnt + w * number_of_priors * cnt + s * cnt;
+                };
```

Comment:
```suggestion
                auto get_idx = [&]() -> uint64_t {
                    return h * layer_width * number_of_priors * 4 + w * number_of_priors * 4 + s * 4;
                };
```

Reply:
good idea, was refactored in more deeply. Thanks